src/main.rs

```diff
@@ -364,16 +365,57 @@ impl LanguageServer for Backend {
         params: GotoDefinitionParams,
     ) -> Result<Option<GotoDefinitionResponse>> {
         info!("ts_query_ls goto_definition: {params:?}");
-        let uri = params.text_document_position_params.text_document.uri;
-        let pos = Position {
-            line: params.text_document_position_params.position.line,
-            character: 0,
+        let uri = &params.text_document_position_params.text_document.uri;
+        let Some(tree) = self.ast_map.get(uri) else {
+            warn!("No AST built for URI: {uri:?}");
+            return Ok(None);
+        };
+        let Some(rope) = self.document_map.get(uri) else {
+            warn!("No document built for URI: {uri:?}");
+            return Ok(None);
         };
-        let range = Range::new(pos, pos);
-        Ok(Some(GotoDefinitionResponse::Scalar(Location {
-            uri,
-            range,
-        })))
+        let cur_pos = params.text_document_position_params.position;
+        let Some(current_node) =
+            get_current_capture_node(tree.root_node(), lsp_position_to_ts_point(cur_pos, &rope))
+        else {
+            return Ok(None);
+        };
+
+        let query = Query::new(&QUERY_LANGUAGE, "(capture) @cap").unwrap();
+        let mut cursor = QueryCursor::new();
+        let provider = TextProviderRope(&rope);
+
+        let mut parser = Parser::new();
+        parser
+            .set_language(&QUERY_LANGUAGE)
+            .expect("Error setting language for Query parser");
+
+        let defs = get_references(
+            &tree.root_node(),
+            &current_node,
+            &query,
+            &mut cursor,
+            &provider,
+            &rope,
+        )
+        .filter(|node| {
+            if let Some(p) = node.parent() {
+                p.kind() != "parameters"
+            } else {
+                true
+            }
+        })
```

Comment:
I think this could just be
```suggestion
        .filter(|node| node.parent().is_none_or(|p| p.kind() != "parameters"))
```